src/codegen.cpp

```diff
@@ -7985,7 +7985,7 @@ static jl_llvm_functions_t
         jl_varinfo_t &varinfo = ctx.slots[i];
         uint8_t flags = jl_array_uint8_ref(src->slotflags, i);
         varinfo.isSA = (jl_vinfo_sa(flags) != 0) || varinfo.isArgument;
-        varinfo.usedUndef = (jl_vinfo_usedundef(flags) != 0) || (!varinfo.isArgument && !src->inferred);
+        varinfo.usedUndef = (jl_vinfo_usedundef(flags) != 0) || !varinfo.isArgument;
```

Comment:
Do we compute this accurately now? It seems the old code assumes that usedundef wasa only valid after inference ran

Reply:
Inference does not touch this flag. Of course, there's also basically no slots, other than arguments in optimized IR.

Reply:
Or make it conservative?
```suggestion
        varinfo.usedUndef = !varinfo.isArgument;
```

Reply:
We assume correctness of this flag in inference. Of course it's possible that our codegen for `varinfo.usedUndef` is wrong, but I don't see any reason to make that assumption. The `->inferred` check here is from all the way back in c5cd44281f6da33c1e58b262058cb0d497f8e9e8 (11 years ago). The frontend is a lot more robust now. Of course, I can't promise there are no bugs, but I think we just have to treat that like any other bug and fix it if it comes up.